<h1><b>TikTok Claims Classification Project </b></h1>

<h3>Capstone Project: Google Advanced Data Analytics Professional Certificate</h3>

<h2><b> Background and Context</b></h2>
<p>TikTok’s user-generated content is vast, diverse, and rapidly growing. With the power given to users to report videos or comments that may contain questionable claims, moderators are faced with an overwhelming volume of reports to investigate. These reports include both factual claims and subjective opinions, and manually sorting them is time-consuming.

To accelerate and optimize the content moderation process, TikTok aims to implement a predictive machine learning model that distinguishes between content that contains a verifiable claim and that which expresses a personal opinion. </p>

<h2><b>Business Impact</b></h2>
<p>Implementing a predictive model to classify user reported content into claims or opinions has direct and measurable business impact</p>
<ul>
    <li>Automating classifications will reduce repsonse time for high priority content.</li>
    <li>Moderators can focus only on claim-based content that may require verification saving time and cost </li>
    <li>Prioritizing actual claims ensures that the company stays ahead in mitigating misinformation and improves user trust</li>
</ul>

<h2><b>Methodology</b></h2>
<h4>1) Data Collection and Preparation </h4>
<ul>
    <li>The goal is to construct a dataframe in Python, perform initial inspection of the provided dataset, and inform TikTok data team members of the findings.</li>
    <li> Understanding the situation:How can the data be prepared and organized</li>
    <li> Understand the data: Creating necessary dataframes to understand the data and also to use in future for EDA and statistical analysis</li>
    <li>Understanding variables:  Using insights from the examination of the summary data to guide deeper investigation into variables</li>
    
</ul>


In [1]:
# Import packages
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 


In [3]:
data = pd.read_csv("tiktok_dataset.csv")

In [4]:
data.head()

,#,claim_status,video_id,video_duration_sec,video_transcription_text,verified_status,author_ban_status,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
0,1,claim,7017666017,59,someone shared with me that drone deliveries a...,not verified,under review,343296.0,19425.0,241.0,1.0,0.0
1,2,claim,4014381136,32,someone shared with me that there are more mic...,not verified,active,140877.0,77355.0,19034.0,1161.0,684.0
2,3,claim,9859838091,31,someone shared with me that american industria...,not verified,active,902185.0,97690.0,2858.0,833.0,329.0
3,4,claim,1866847991,25,someone shared with me that the metro of st. p...,not verified,active,437506.0,239954.0,34812.0,1234.0,584.0
4,5,claim,7105231098,19,someone shared with me that the number of busi...,not verified,active,56167.0,34987.0,4110.0,547.0,152.0


We can see that the data contains a mix of categorical, text and numerical data. Each row in the data represents the claim or opinion along with other metadata about the video. 

In [7]:
# get summary info
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19382 entries, 0 to 19381
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   #                         19382 non-null  int64  
 1   claim_status              19084 non-null  object 
 2   video_id                  19382 non-null  int64  
 3   video_duration_sec        19382 non-null  int64  
 4   video_transcription_text  19084 non-null  object 
 5   verified_status           19382 non-null  object 
 6   author_ban_status         19382 non-null  object 
 7   video_view_count          19084 non-null  float64
 8   video_like_count          19084 non-null  float64
 9   video_share_count         19084 non-null  float64
 10  video_download_count      19084 non-null  float64
 11  video_comment_count       19084 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 1.8+ MB


There are a total of 19382 rows and 11 columns in the dataset with each column having datatype of either int64, object or float64. There are missing values in columns claim_status, video_transcription_text, video_view_count, video_like_count, video_share_count, video_download_count and video_comment_count.

In [6]:
# get summary statistics
data.describe()

,#,video_id,video_duration_sec,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
count,19382.000000,1.938200e+04,19382.000000,19084.000000,19084.000000,19084.000000,19084.000000,19084.000000
mean,9691.500000,5.627454e+09,32.421732,254708.558688,84304.636030,16735.248323,1049.429627,349.312146
std,5595.245794,2.536440e+09,16.229967,322893.280814,133420.546814,32036.174350,2004.299894,799.638865
min,1.000000,1.234959e+09,5.000000,20.000000,0.000000,0.000000,0.000000,0.000000
25%,4846.250000,3.430417e+09,18.000000,4942.500000,810.750000,115.000000,7.000000,1.000000
50%,9691.500000,5.618664e+09,32.000000,9954.500000,3403.500000,717.000000,46.000000,9.000000
75%,14536.750000,7.843960e+09,47.000000,504327.000000,125020.000000,18222.000000,1156.250000,292.000000
max,19382.000000,9.999873e+09,60.000000,999817.000000,657830.000000,256130.000000,14994.000000,9599.000000
